amazon-kinesis-client/src/test/java/software/amazon/kinesis/common/StreamARNUtilTest.java

```diff
@@ -0,0 +1,70 @@
+package software.amazon.kinesis.common;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.MockitoAnnotations;
+import org.powermock.api.mockito.PowerMockito;
+import org.powermock.core.classloader.annotations.PrepareForTest;
+import org.powermock.modules.junit4.PowerMockRunner;
+import software.amazon.awssdk.arns.Arn;
+import software.amazon.awssdk.regions.Region;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+
+@RunWith(PowerMockRunner.class)
+@PrepareForTest({ StreamARNUtil.class })
+public class StreamARNUtilTest {
+    private static final String STS_RESPONSE_ARN_FORMAT = "arn:aws:sts::%s:assumed-role/Admin/alias";
+    private static final String KINESIS_STREAM_ARN_FORMAT = "arn:aws:kinesis:us-east-1:%s:stream/%s";
+
+    @Before
+    public void setUp() {
+        MockitoAnnotations.initMocks(this);
+        PowerMockito.spy(StreamARNUtil.class);
+    }
+
+    @Test
+    public void testGetStreamARNHappyCase() throws Exception {
+        String streamName = "testStreamName";
+        String accountId = "123456789012";
+        PowerMockito.doReturn(Optional.of(Arn.fromString(String.format(STS_RESPONSE_ARN_FORMAT, accountId))))
+                .when(StreamARNUtil.class, "getStsCallerArn");
+
+        Optional<Arn> actualStreamARNOptional = StreamARNUtil.getStreamARN(streamName, Region.US_EAST_1);
+        String expectedStreamARN = String.format(KINESIS_STREAM_ARN_FORMAT, accountId, streamName);
+        assertTrue(actualStreamARNOptional.isPresent());
+        assertEquals(expectedStreamARN, actualStreamARNOptional.get().toString());
```

Comment:
If you change to use a mock `StsClient` object, then you should validate expectations on that object (in this test, and all other tests). Example:
```
Mockito.verify(mockStsClient).getCallerIdentity();
```

Validating the mock helps ensure any `when(...)` expectations were triggered, and reduces the likelihood that the test passes due to some other fortunate happenstance.